server/channels/app/plugin_install_test.go

```diff
@@ -265,6 +266,69 @@ func TestInstallPluginLocally(t *testing.T) {
 			assertBundleInfoManifests(t, th, []*model.Manifest{manifest})
 		})
 	})
+
+	installPluginUpdateConfig := func(t *testing.T, th *TestHelper, id, version string, installationStrategy pluginInstallationStrategy) (*model.Manifest, map[string]any, *model.AppError) {
+		t.Helper()
+		var mp map[string]any
+
+		manifest := &model.Manifest{
+			Id:      id,
+			Version: version,
+		}
+		manifestJSON, jsonErr := json.Marshal(manifest)
+		require.NoError(t, jsonErr)
+		reader := makeInMemoryGzipTarFile(t, []testFile{
+			{"plugin.json", string(manifestJSON)},
+		})
+
+		actualManifest, appError := th.App.ch.installPluginLocally(reader, installationStrategy)
+		if actualManifest != nil {
+			require.Equal(t, manifest, actualManifest)
+		}
+
+		th.App.ch.cfgSvc.UpdateConfig(func(cfg *model.Config) {
+			if _, ok := cfg.PluginSettings.Plugins[actualManifest.Id]; !ok {
+				cfg.PluginSettings.Plugins[actualManifest.Id] = make(map[string]any)
+
+				for _, pluginSetting := range manifest.SettingsSchema.Settings {
+					cfg.PluginSettings.Plugins[actualManifest.Id][strings.ToLower(pluginSetting.Key)] = pluginSetting.Default
+				}
+			}
+			mp = cfg.PluginSettings.Plugins[actualManifest.Id]
+		})
```

Comment:
I don't understand what you are doing there. The code we are testing does set the default values, not this test.

You can just use `th.App.Config().PluginSettings.Plugins[actualManifest.Id]` to get the plugin settings stored in the config after the test installed the plugin.

Maybe you need to use `setupPluginAPITest` to install a plugin with a custom manifest. Check out https://github.com/mattermost/mattermost/blob/34fdcf80c374cd87a33185db75b39b6dafd04a1a/server/channels/app/integration_action_test.go#L593-L626 for an example usage.

Reply:
So.... the point of `setupPluginAPITest()` is to just create a dummy plugin to test against, the error from the server will not matter, right?

I wonder why it demonstrates creating a server when you could just run the `main()` func?